infra/modules/database/main.tf

```diff
@@ -0,0 +1,362 @@
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "db_backup_role_policy_attachment" {
+  role       = aws_iam_role.db_backup_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSBackupServiceRolePolicyForBackup"
+}
+
+#----------------------------------#
+# IAM role for enhanced monitoring #
+#----------------------------------#
+
+resource "aws_iam_role" "rds_enhanced_monitoring" {
+  name_prefix        = "${var.name}-enhanced-monitoring-"
+  assume_role_policy = data.aws_iam_policy_document.rds_enhanced_monitoring.json
+}
+
+resource "aws_iam_role_policy_attachment" "rds_enhanced_monitoring" {
+  role       = aws_iam_role.rds_enhanced_monitoring.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonRDSEnhancedMonitoringRole"
+}
+
+data "aws_iam_policy_document" "rds_enhanced_monitoring" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["monitoring.rds.amazonaws.com"]
+    }
+  }
+}
+
+#---------------#
+# Query Logging #
+#---------------#
+
+resource "aws_rds_cluster_parameter_group" "rds_query_logging" {
+  name        = var.name
+  family      = "aurora-postgresql13"
+  description = "Default cluster parameter group"
+
+  parameter {
+    name = "log_statement"
+    # Logs data definition statements (e.g. DROP, ALTER, CREATE)
+    value = "ddl"
+  }
+
+  parameter {
+    name = "log_min_duration_statement"
+    # Logs all statements that run 100ms or longer
+    value = "100"
+  }
+}
+
+#-----------------------------------------------------------------------------#
+# Role Manager Lambda Function                                                #
+#                                                                             #
+# Resources for the lambda function that is used for managing database roles  #
+# This includes creating and granting permissions to roles                    #
+# as well as viewing existing roles                                           #
+#-----------------------------------------------------------------------------#
+
+resource "aws_lambda_function" "role_manager" {
+  function_name = local.role_manager_name
+
+  filename         = local.role_manager_package
+  source_code_hash = data.archive_file.role_manager.output_base64sha256
+  runtime          = "python3.9"
+  handler          = "role_manager.lambda_handler"
+  role             = aws_iam_role.role_manager.arn
+  kms_key_arn      = aws_kms_key.role_manager.arn
+
+  # Only allow 1 concurrent execution at a time
+  reserved_concurrent_executions = 1
+
+  vpc_config {
+    subnet_ids         = var.private_subnet_ids
+    security_group_ids = [aws_security_group.role_manager.id]
+  }
+
+  environment {
+    variables = {
+      DB_HOST       = aws_rds_cluster.db.endpoint
+      DB_PORT       = aws_rds_cluster.db.port
+      DB_USER       = local.master_username
+      DB_PASSWORD   = aws_ssm_parameter.random_db_password.value
```

Comment:
This value is displayed in plaintext visible in the aws console > function > configuration. I think we should lock this down by having the function [retrieve the password from SSM](https://aws.amazon.com/blogs/compute/sharing-secrets-with-aws-lambda-using-aws-systems-manager-parameter-store/).

Reply:
Or create an issue to lock this down.

Reply:
Good catch thanks